tests/testbugs.nim

```diff
@@ -98,6 +98,40 @@ suite "Asynchronous issues test suite":
 
     result = r1 and r2
 
+  proc createBigMessage(size: int): seq[byte] =
+    var message = "MESSAGE"
+    var res = newSeq[byte](size)
+    for i in 0 ..< len(result):
+      res[i] = byte(message[i mod len(message)])
+    res
+
+  proc testIndexError(): Future[bool] {.async.} =
```

Comment:
Thanks for finding and constructing a reproducible test case for this elusive bug!!